### PR TITLE
Story (IFrame) doc block not refreshing while navigating different stories

### PR DESCRIPTION
**Pull request by matheo**
Issue:

I have a default MDX parameters.docs.page like this:
```
import { Description, Props, Story, Subtitle, Title } from '@storybook/addon-docs/blocks';

<Title />
<Subtitle />
<Description />

<Story id="." />

<Props of="." />
```
but the Story iframe (with angular framework) is not refreshing the content while I navigate different stories, so I had to dig into the problem and found the cause.

## What I did

Detected `src` change in the `IFrame` component, and also catched a problem with the `Story` doc block not extracting the `docs` values from the parameters, and trying to get them from the StoryProps directly.

Also, I exposed some common components stuff to be usable from customized components, so third-party developers can reuse the `EmptyBlock` and `ZoomContext` if needed.

## How to test

- Is this testable with Jest or Chromatic screenshots?
No
- Does this need a new example in the kitchen sink apps?
No
- Does this need an update to the documentation?
No

### CODE REVIEW

File: lib/components/src/blocks/IFrame.tsx [hunk 1]
```diff
@@ -27,7 +27,10 @@ export class IFrame extends Component<IFrameProps> {
   }
 
   shouldComponentUpdate(nextProps: IFrameProps) {
-    const { scale } = nextProps;
+    const { scale, src } = nextProps;
```
Comment (shilman) [inline lib/components/src/blocks/IFrame.tsx:30]: @ndelangen can you look at this?

File: lib/components/src/blocks/index.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
+export * from './EmptyBlock';
+export * from './ZoomContext';
```
Comment (shilman) [inline lib/components/src/blocks/index.ts:2]: Why export these?

File: addons/docs/src/blocks/Story.tsx [hunk 3]
```diff
@@ -53,8 +53,8 @@ export const getStoryProps = (props: StoryProps, context: DocsContextProps): Pur
   const data = context.storyStore.fromId(previewId) || {};
 
   const { height, inline } = props;
-  const { parameters = {}, docs = {} } = data;
-  const { framework = null } = parameters;
+  const { parameters = {} } = data;
+  const { docs = {}, framework = null } = parameters;
```
Comment (shilman) [inline addons/docs/src/blocks/Story.tsx:57]: This looks right 👍

File: lib/components/src/blocks/index.ts (hunk 2 above)
Comment (matheo) [reply lib/components/src/blocks/index.ts:2]: I needed them while trying to reproduce the issue with custom components.
It doesn't hurt to share them, but I could remove that commit from the PR if needed.

File: lib/components/src/blocks/IFrame.tsx (hunk 1 above)
Comment (matheo) [reply lib/components/src/blocks/IFrame.tsx:30]: The way to reproduce this is described in #10851
that granular MDX docpage shows how the Story angular-iframe is not refreshed

Comment (ndelangen) [reply lib/components/src/blocks/IFrame.tsx:30]: I think it'd be better to set a key on the iframe based on the storyId.

If the key changes, react will unmount & create a new element.

I think this totally defeats the purpose of the `shouldComponentUpdate`

**Review by ndelangen — CHANGES_REQUESTED** [review]

File: lib/components/src/blocks/IFrame.tsx (hunk 1 above)
Comment (matheo) [reply lib/components/src/blocks/IFrame.tsx:30]: @ndelangen the StoryId is inside `nextProps.id`.
I'm not a React developer so I'm not familiar at all with the lifecycle.

**Comment by ndelangen** [comment]
@matheo want to pair on this perhaps to get this across the line?

**Pull request closed without merging**